vyper/parser/expr.py

```diff
@@ -1035,3 +999,62 @@ def parse_variable_location(cls, expr, context):
         if not o.location:
             raise StructureException("Looking for a variable location, instead got a value", expr)
         return o
+
+
+def parse_sequence(base_node, elements, context):
+    """
+    Generate an LLL node from a sequence of Vyper AST nodes, such as values inside a
+    list/tuple or arguments inside a call.
+
+    Arguments
+    ---------
+    base_node : VyperNode
+        Parent node which contains the sequence being parsed.
+    elements : List[VyperNode]
+        A list of nodes within the sequence.
+    context : Context
+        Currently active local context.
+
+    Returns
+    -------
+    List[LLLNode]
+        LLL nodes that must execute prior to generating the actual sequence in order to
+        avoid memory corruption issues. This list may be empty, depending on the values
+        within `elements`.
+    List[LLLNode]
+        LLL nodes which collectively represent `elements`.
+    """
+    init_lll = []
+    sequence_lll = []
+    for node in elements:
+        if isinstance(node, vy_ast.List):
+            init, seq = parse_sequence(node, node.elements, context)
```

Comment:
Add a comment here "we must make sure that every layer deep pushes all the computations into `init`" or something